### PR TITLE
Add missing urls on join_paths

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

File: src/libstd/env.rs [hunk 1]
```diff
@@ -400,15 +400,19 @@ pub struct JoinPathsError {
...
-/// Joins a collection of `Path`s appropriately for the `PATH`
+/// Joins a collection of [`Path`]s appropriately for the `PATH`
 /// environment variable.
 ///
-/// Returns an `OsString` on success.
+/// Returns an [`OsString`] on success.
 ///
-/// Returns an `Err` (containing an error message) if one of the input
-/// `Path`s contains an invalid character for constructing the `PATH`
+/// Returns an [`Err`][err] (containing an error message) if one of the input
```
Comment (frewsxcv) [inline src/libstd/env.rs]: I think you could have also just done:

```
[`Err`][]

[`Err`]: https://foo.bar
```

Comment (frewsxcv) [inline src/libstd/env.rs]: But what you have is fine :+1:

Comment (GuillaumeGomez) [inline src/libstd/env.rs]: I find it a bit less clear. 😉

**Pull request merged**